### PR TITLE
function to get acme user for routes

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:

- Add a function to get an acme user for routes
- add acme_user_v2 models

## Security considerations

None

### CODE REVIEW

**Review by ccostino — COMMENTED** [review]
Awesome, thanks @bengerman13!

Everything looks right to me and I only had a couple of specific questions, and I'd love to walk through this together too if you don't mind, just to better understand what's happening as this is uncharted territory for me. :-)

File: renewer/tasks/letsencrypt.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return
+
+    route.acme_user = acme_user
+
+    key = josepy.JWKRSA(
+        key=rsa.generate_private_key(
+            public_exponent=65537, key_size=2048, backend=default_backend()
+        )
+    )
+    private_key_pem_in_binary = key.key.private_bytes(
```
[the remark this answers is not in the document]
Comment (bengerman13) [reply renewer/tasks/letsencrypt.py:39]: probably worth noting that I took this wholesale from the external-domain-broker 🤣

File: renewer/tasks/letsencrypt.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+    acme_user.private_key_pem = private_key_pem_in_binary.decode("utf-8")
+
+    net = client.ClientNetwork(key, user_agent="cloud.gov legacy domain renewer")
+    directory = messages.Directory.from_json(net.get(config.ACME_DIRECTORY).json())
+    client_acme = AcmeClient(directory, net=net)
+
+    acme_user.email = "cloud-gov-operations@gsa.gov"
```
[the remark this answers is not in the document]
Comment (bengerman13) [reply renewer/tasks/letsencrypt.py:50]: I suppose. It'd probably also make it a little easier for anyone who wants to borrow this

[the remark this answers is not in the document]
Comment (bengerman13) [reply renewer/tasks/letsencrypt.py]: Added this as a config option

**Review by ccostino — APPROVED** [review]
Thanks, @bengerman13!

**Pull request merged**